Playground/android-use-agent/app/src/main/java/com/runanywhere/agent/kernel/AgentKernel.kt

```
@@ -0,0 +1,770 @@
+                if (System.currentTimeMillis() - startTime > MAX_DURATION_MS) {
+                    emit(AgentEvent.Done("Max duration reached"))
+                    return@flow
+                }
+
+                delay(STEP_DELAY_MS)
+            }
+
+            emit(AgentEvent.Speak("I've reached the maximum steps."))
+            emit(AgentEvent.Done("Max steps reached"))
+
+        } catch (e: CancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)